Add split-layout content and example stubs (#266)
vaadin/docs: articles/ds/components/split-layout/index.asciidoc

= Split Layout

// tag::description[]
Split Layout is a component with two content areas and a draggable split handle between them.
// end::description[]

.Work In Progress
[NOTE]
====
The usage examples for this component are currently in progress.
In the meantime, you can view the previous examples for the web component and the Java/Flow component.

[.buttons]
- https://vaadin.com/components/vaadin-split-layout/html-examples[Web Component Examples]
- https://vaadin.com/components/vaadin-split-layout/java-examples[Java Component Examples]
====

[.example]
--

[source,typescript]
----
include::{root}/frontend/demo/component/splitlayout/split-layout-basic.ts[render,tags=snippet,indent=0,group=TypeScript]
----

[source, java]
----
include::{root}/src/main/java/com/vaadin/demo/component/splitlayout/SplitLayoutBasic.java[render,tags=snippet,indent=0,group=Java]
----
--

== Orientation

The default orientation is horizontal, placing the content areas side by side.
The orientation can also be vertical.

Orientation should be set based on the content and the screen size.

> Alternatively, allow the user to choose which orientation they wish to use.

Reviewer: ```suggestion
The user can also be allowed to choose which orientation they wish to use.
```

[.example]
--

[source,typescript]
----
include::{root}/frontend/demo/component/splitlayout/split-layout-orientation.ts[render,tags=snippet,indent=0,group=TypeScript]
----

[source, java]
----
include::{root}/src/main/java/com/vaadin/demo/component/splitlayout/SplitLayoutOrientation.java[render,tags=snippet,indent=0,group=Java]
----
--

== Splitter Position

The initial splitter position is determined by the default size of the two content area components.
Their height and width affect the position when using a vertical and horizontal orientation, respectively.

The initial split position can also be explicitly set using a percentage value.

[.example]
--

[source,typescript]
----
include::{root}/frontend/demo/component/splitlayout/split-layout-initial-splitter-position.ts[render,tags=snippet,indent=0,group=TypeScript]
----

[source, java]
----
include::{root}/src/main/java/com/vaadin/demo/component/splitlayout/SplitLayoutInitialSplitterPosition.java[render,tags=snippet,indent=0,group=Java]
----
--

> The splitter will respect the minimum and maximum size of the content area components.

Reviewer: ```suggestion
The splitter respects the minimum and maximum size of the content area components.
```

[.example]
--

[source,typescript]
----
include::{root}/frontend/demo/component/splitlayout/split-layout-min-max-size.ts[render,tags=snippet,indent=0,group=TypeScript]
----

[source, java]
----
include::{root}/src/main/java/com/vaadin/demo/component/splitlayout/SplitLayoutMinMaxSize.java[render,tags=snippet,indent=0,group=Java]
----
--

The split can be adjusted programmatically, for example, by using a Button.

> This is useful when the user wants to quickly toggle between certain positions.

Reviewer: ```suggestion
This is useful when the user wants to toggle between certain positions.
```

[.example]
--

[source,typescript]
----
include::{root}/frontend/demo/component/splitlayout/split-layout-toggle.ts[render,tags=snippet,indent=0,group=TypeScript]
----

[source, java]
----
include::{root}/src/main/java/com/vaadin/demo/component/splitlayout/SplitLayoutToggle.java[render,tags=snippet,indent=0,group=Java]
----
--

== Theme Variants

There are two theme variants: `small` and `minimal`.

[.example]
--

[source,typescript]
----
include::{root}/frontend/demo/component/splitlayout/split-layout-theme-variants.ts[render,tags=snippet,indent=0,group=TypeScript]
----

[source, java]
----
include::{root}/src/main/java/com/vaadin/demo/component/splitlayout/SplitLayoutThemeVariants.java[render,tags=snippet,indent=0,group=Java]
----
--

The `small` theme variant makes the divider smaller.

> The `minimal` theme variant removes the divider completely.

Reviewer: ```suggestion
The `minimal` theme variant hides the visual divider.
```

> Both variants only show the split handle on hover and are therefore not ideal for touch devices.

Reviewer: ```suggestion
Both variants only show the split handle on hover and are not ideal for touch devices.
```

[.example]
--

[source,typescript]
----
include::{root}/frontend/demo/component/splitlayout/split-layout-minimal-theme-variants.ts[render,tags=snippet,indent=0,group=TypeScript]
----

[source, java]
----
include::{root}/src/main/java/com/vaadin/demo/component/splitlayout/SplitLayoutMinimalThemeVariants.java[render,tags=snippet,indent=0,group=Java]
----
--

When using the `minimal` theme variant, it is recommended to somehow suggest the split between the two sides, for example by styling one side as a layer on top of the other.

> While these variants reduce the visual clutter, they make it less obvious to the user that the content is resizable.

Reviewer: Add `resizable` to [.github/styles/Vocab/Docs/accept.txt](https://github.com/vaadin/docs/blob/latest/.github/styles/Vocab/Docs/accept.txt)

Reviewer 2: Done.
